src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -4140,6 +4156,33 @@ theorem getLsbD_twoPow (i j : Nat) : (twoPow w i).getLsbD j = ((i < w) && (i = j
           simp at hi
         simp_all
 
+@[simp]
+theorem msb_twoPow {i w: Nat} :
+    (twoPow w i).msb = (decide (i < w) && decide (i = w - 1)) := by
+  simp only [BitVec.msb, getMsbD_eq_getLsbD, Nat.sub_zero, getLsbD_twoPow,
+    Bool.and_iff_right_iff_imp, Bool.and_eq_true, decide_eq_true_eq, and_imp]
+  intros
+  omega
```

Comment:
```suggestion
  simp [BitVec.msb, getMsbD_eq_getLsbD]
  omega
```

Reply:
thank you, but this proof was upstreamed already - I just moved the theorem up (was previously at line 4156). I would suggest opening a new PR with this change, to avoid unnecessary complexity in this PR.

Reply:
Fine with me. Can you close this comment then?